app/models/sensor.rb

```diff
@@ -2,6 +2,13 @@ class Sensor < ActiveRecord::Base
   has_many :readings, dependent: :destroy
   has_many :measurement_sensors, dependent: :destroy
   has_many :measurements, through: :measurement_sensors
+  has_many :locations, dependent: :destroy
 
-  validates :long, :lat, presence: true
+  before_create :generate_token
+
+  private
+
+  def generate_token
+    self.token = SecureRandom.urlsafe_base64.gsub('-_','')[0,12]
```

Comment:
Do you know if I can shorten its length to 12? The github issue discussion indicates this option was removed. I need it relatively short as people might need to type that in on their mobiles.